### PR TITLE
Fix failing static analyzer tests

### LINKED ISSUE #7074: It is not possible to use the analysistest package with Bazel
# 💎 Issue

### Description

Trying to run any analyzer tests such as these with Bazel

```
func TestAnalyzer(t *testing.T) {
	analysistest.Run(t, analysistest.TestData(), Analyzer)
}
```

results in

```
Failed
=== RUN   TestAnalyzer
    TestAnalyzer: analysistest.go:103: go tool not available: exec: "go": executable file not found in $PATH
--- FAIL: TestAnalyzer (0.00s)
```

`go test` works perfectly fine.

Because of this problem it is not possible to run such tests during the CI process. Example: https://github.com/prysmaticlabs/prysm/blob/master/tools/analyzers/cryptorand/analyzer_test.go (the file is explicitly ignored in `BAZEL.build`).

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Tools

**What does this PR do? Why is it needed?**

This PR fixes failing tests by adding/modifying expected diagnostics and adds a readme file about running analyzer tests.

**Which issues(s) does this PR fix?**

Closes #7074 

**Other notes for review**

N/A

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -46,6 +46,9 @@ jobs:
         # Use blst tag to allow go and bazel builds for blst.
         run: go build -v --tags=blst_enabled ./...
 
+      - name: Test static analyzers
```
Comment (farazdagi) [inline .github/workflows/go.yml:49]:
```suggestion
      - name: Run static analyzers
```

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -46,6 +46,9 @@ jobs:
...
+        run: go test -v ./tools/analyzers/...
```
Comment (farazdagi) [inline .github/workflows/go.yml:50]: Also, do static analyzers need to be w/i `Build`? I mean, if static analyzers find some problems it is not build failed, it is some test failed. Can we probably move it into `Check` (or its own `Test` section)? Are there any cons?

**Review by farazdagi — CHANGES_REQUESTED** [review]

File: .github/workflows/go.yml (hunk 1 above)
Comment (rkapka) [reply .github/workflows/go.yml:49]: Analyzer checks are already a part of the CI bazel build. This PR enables unit testing these analyzers during CI, so we're not running analyzers here.

File: .github/workflows/go.yml (hunk 2 above)
Comment (rkapka) [reply .github/workflows/go.yml:50]: I agree that it would be better to move these tests into a separate `Test` section.

**Review by farazdagi — DISMISSED** [review]
Looks good, thank you!

**Review by farazdagi — APPROVED** [review]

**Pull request merged**